File: TP_01/Ciframento/cesar.java
import java.util.Scanner;
    
public class cesar{

public static String cifra (String palavra){
int tamanho = palavra.length();
char letra  = ' ';
String s = "";
for(int x = 0; x < tamanho; x++){
    letra = (char)(palavra.charAt(x)+3);
    s = s + letra;
}
return(s);
}
public static boolean fim(String palavra){
    boolean ok = false;
    if(palavra.charAt(0) == 'F' && palavra.charAt(1) == 'I' && palavra.charAt(2) == 'M'){
        ok = true;
    }
    return(ok);
}


public static void main(String[] args){
    Scanner sc = new Scanner(System.in);
    String palavra = "";
    palavra = sc.nextLine();

    while(!fim(palavra)){
        System.out.println(cifra(palavra));
        palavra = sc.nextLine();
    }
    sc.close();
}
}

File: Lab_Aeds2/TP_01a/CifraR/cesar.c
#include <stdio.h>
#include <stdbool.h>

char* cifra(char palavra[], int n, int tamanho){
    if(n < tamanho){
        palavra[n] = (char)(palavra[n] + 3);
        return cifra(palavra, n + 1, tamanho);
    }
    return palavra;
}

bool fim(char palavra[]){
    bool ok = false;
    if(palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M' && palavra[3] == '\0'){
        ok = true;
    }
    return ok;
}

int main(){
    char palavra[1000];
    fgets(palavra, 1000, stdin);
    int tamanho = 0;
    while(palavra[tamanho] != '\0' && palavra[tamanho] != '\n'){
        tamanho++;
    }
    palavra[tamanho] = '\0';

    while(!fim(palavra)){
        printf("%s\n", cifra(palavra, 0, tamanho-1));
        fgets(palavra, 1000, stdin);
        tamanho = 0;
        while(palavra[tamanho] != '\0' && palavra[tamanho] != '\n'){
            tamanho++;
        }
        palavra[tamanho] = '\0';
    }

    return 0;
}
